jdk-1.7-parent/wicketstuff-lazymodel/src/main/java/org/wicketstuff/lazymodel/LazyModel.java

```diff
@@ -556,6 +556,33 @@ private void fillArguments(Object[] args) {
 	}
 
 	/**
+	 * Shortcut for {@link #from(Class)} and {@link #bind(Object)}. Use this for quick model building on top of a
+	 * type-erased model.
+	 *
+	 * @param target The model to bind to.
+	 * @param type The type parameter for that model.
+	 * @param <T>
+	 * @return a result proxy for further evaluation.
+	 * @see #fromObject(IModel)
+	 */
+	public static <T> T from(IModel<T> target, Class<T> type) {
+		return (T) new BoundEvaluation<T>(type, target).proxy();
+	}
+
+	/**
+	 * Start a lazy evaluation, using the reflected runtime type of the given model's object. Use this for terse model
+	 * building on top of a type erased model.
+	 * @param target The model to bind to.
+	 * @param <T>
+	 * @return a result proxy for further evaluation.
+	 * @see #from(IModel, Class)
+	 */
+	public static <T> T fromObject(IModel<T> target)
+	{
+		return (T) new BoundEvaluation<T>(target.getObject().getClass(), target).proxy();
```

Comment:
Yes "eagerlyFrom" would definitely go into the right direction:

I find it very important that LazyModel never accesses the model object during evaluation - I'm afraid adding such a possibility will lead to people calling the wrong method.

Which model implementations are you using in your project, so you have to provide the object class by yourself?